script.io.benchmark/default.py

```diff
@@ -0,0 +1,154 @@
+ #   Copyright (C) 2017 Lunatixz
+#
+#
+# This file is part of I/O Benchmark.
+#
+# I/O Benchmark is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# I/O Benchmark is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with I/O Benchmark.  If not, see <http://www.gnu.org/licenses/>.
+
+import os, sys, monkeytest
+import xbmc, xbmcgui, xbmcplugin, xbmcvfs, xbmcaddon
+
+
+# Plugin Info
+ADDON_ID       = 'script.io.benchmark'
+REAL_SETTINGS  = xbmcaddon.Addon(id=ADDON_ID)
+ADDON_NAME     = REAL_SETTINGS.getAddonInfo('name')
+ADDON_PATH     = (REAL_SETTINGS.getAddonInfo('path').decode('utf-8'))
+ADDON_VERSION  = REAL_SETTINGS.getAddonInfo('version')
+SETTINGS_LOC   = REAL_SETTINGS.getAddonInfo('profile')
+ICON           = os.path.join(ADDON_PATH, 'icon.png')
+FANART         = os.path.join(ADDON_PATH, 'fanart.jpg') 
+                        
+ADDON_FILE   = os.path.join((xbmc.translatePath(ADDON_PATH)),'test.tmp')
+SETTINGS_FILE= os.path.join((xbmc.translatePath(SETTINGS_LOC)),'test.tmp')
+
+if xbmcvfs.exists(SETTINGS_LOC) == False:
+    xbmcvfs.mkdirs(SETTINGS_LOC)
+     
+TEST_PATHS = None
+# change the constants below according to your needs
+WRITE_MB       = 128    # total MBs Wrote during test
+WRITE_BLOCK_KB = 1024   # KBs in each write block
+READ_BLOCK_B   = 512    # bytes in each read block (high values may lead to
+                        # invalid results because of system I/O scheduler        
+                        # file must be at drive under test
+  
+def repeat_to_length(string_to_expand, length):
+   return (string_to_expand * ((length/len(string_to_expand))+1))[:length]
+ 
+def log(msg, level = xbmc.LOGDEBUG):
+    xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + str(msg), level)
+    
+def showText(text, header=ADDON_NAME):
+    xbmcgui.Dialog().textviewer(header, text)
+     
+def okay(line1= '', line2= '', line3= '', header=ADDON_NAME):
+    xbmcgui.Dialog().ok(header, line1, line2, line3)
+   
+try:
```

Comment:
as far as I'm aware, it is **always** recommended to use the `if __name__ == '__main__':` clause as your entry point for execution (no matter if it's a file not meant to be imported or not).
Think of other languages like java, c++ etc. all have their way of defining the "main entry point" for what ever needs to be run.

Just because python allows us to have this kind of floating code doesn't mean it's good practices :)

now. In python we work module-based which means each *other* python file (or directory containing more python code) is considered a module.
These typically do **not** have this entry point as they usually are only used via imports (but at the same time, every piece of code-**logic** defined is within classes/methods).

[as an example for *floating* code](https://github.com/linuxwhatelse/plugin.audio.linuxwhatelse.gmusic/blob/master/addon/__init__.py) (couldn't think of another example except one of my addons) a `__init__.py` (declaring the **addon** directory as python module) may very well have some form of *floating* (no clue how to properly call that... I should research that at some point) code for declaration purposes.

**TL;DR**
Encapsulate as much logic within methods and you're fine. :)